src/components/ColumnSwitcher/index.js

```diff
@@ -72,48 +85,24 @@ class ColumnSwitcher extends Component {
         >
           :
         </div>
+
         {this.state.visible && (
           <div
             className="Sticky-React-Table--Header-Column-Switcher-Dropdown"
             ref={this.handleMenuRef}
           >
-            {columns
-              .filter(({ isCheckbox }) => !isCheckbox)
-              .map(({ title, dataKey, visible: isChecked }, index) => {
-                const checkbox = (
-                  <label htmlFor={dataKey}>
-                    <input
-                      type="checkbox"
-                      id={dataKey}
-                      name="column"
-                      onChange={onChange}
-                      checked={isChecked}
-                    />
-                    {title}
-                  </label>
-                );
-
-                return (
-                  <div
-                    className="Sticky-React-Table--Header-Column-Switcher-Item"
-                    key={title || dataKey || index}
-                  >
-                    {renderElement(
-                      checkboxRenderer,
-                      {
-                        checkbox,
-                        id: dataKey,
-                        dataKey,
-                        onChange,
-                        isChecked,
-                        title,
-                        type: 'columnSwitcher'
-                      },
-                      checkbox
-                    )}
-                  </div>
-                );
-              })}
+            {columns.map(({ title, dataKey, visible: isChecked }) => (
+              <ColumnSwitcherItem
+                key={dataKey}
+                {...{
```

Comment:
Maybe we should create a variable and spread that here, because I feel this is a bit difficult to read, but we could always do this later.